<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>863</width>
    <height>472</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>863</width>
    <height>472</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>863</width>
    <height>472</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Supercomputer</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QPushButton" name="simulateButton">
    <property name="geometry">
     <rect>
      <x>280</x>
      <y>380</y>
      <width>101</width>
      <height>51</height>
     </rect>
    </property>
    <property name="cursor">
     <cursorShape>PointingHandCursor</cursorShape>
    </property>
    <property name="text">
     <string>Run simulation</string>
    </property>
   </widget>
   <widget class="QGroupBox" name="userGroup">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>20</y>
      <width>171</width>
      <height>71</height>
     </rect>
    </property>
    <property name="title">
     <string>Number of users</string>
    </property>
    <property name="checkable">
     <bool>false</bool>
    </property>
    <widget class="QSpinBox" name="constantUsers">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>30</y>
       <width>71</width>
       <height>23</height>
      </rect>
     </property>
     <property name="showGroupSeparator" stdset="0">
      <bool>true</bool>
     </property>
     <property name="maximum">
      <number>1000</number>
     </property>
    </widget>
    <widget class="QLabel" name="fromUsersLabel">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>20</y>
       <width>28</width>
       <height>14</height>
      </rect>
     </property>
     <property name="text">
      <string>From</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="fromUsers">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>40</y>
       <width>71</width>
       <height>23</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="toUsersLabel">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>20</y>
       <width>13</width>
       <height>14</height>
      </rect>
     </property>
     <property name="text">
      <string>To</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="toUsers">
     <property name="geometry">
      <rect>
       <x>90</x>
       <y>40</y>
       <width>71</width>
       <height>23</height>
      </rect>
     </property>
     <property name="minimum">
      <number>1</number>
     </property>
     <property name="maximum">
      <number>1000</number>
     </property>
    </widget>
    <widget class="QCheckBox" name="randomUsers">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>0</y>
       <width>71</width>
       <height>16</height>
      </rect>
     </property>
     <property name="cursor">
      <cursorShape>PointingHandCursor</cursorShape>
     </property>
     <property name="text">
      <string>random</string>
     </property>
    </widget>
    <zorder>toUsers</zorder>
    <zorder>fromUsersLabel</zorder>
    <zorder>fromUsers</zorder>
    <zorder>constantUsers</zorder>
    <zorder>toUsersLabel</zorder>
    <zorder>randomUsers</zorder>
   </widget>
   <widget class="QGroupBox" name="jobsGroup">
    <property name="geometry">
     <rect>
      <x>210</x>
      <y>20</y>
      <width>171</width>
      <height>71</height>
     </rect>
    </property>
    <property name="title">
     <string>Number of jobs</string>
    </property>
    <property name="checkable">
     <bool>false</bool>
    </property>
    <widget class="QSpinBox" name="constantJobs">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>30</y>
       <width>71</width>
       <height>23</height>
      </rect>
     </property>
     <property name="showGroupSeparator" stdset="0">
      <bool>true</bool>
     </property>
     <property name="maximum">
      <number>1000</number>
     </property>
    </widget>
    <widget class="QLabel" name="fromJobsLabel">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>20</y>
       <width>28</width>
       <height>14</height>
      </rect>
     </property>
     <property name="text">
      <string>From</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="fromJobs">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>40</y>
       <width>71</width>
       <height>23</height>
      </rect>
     </property>
     <property name="maximum">
      <number>1000</number>
     </property>
    </widget>
    <widget class="QLabel" name="toJobsLabel">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>20</y>
       <width>13</width>
       <height>14</height>
      </rect>
     </property>
     <property name="text">
      <string>To</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="toJobs">
     <property name="geometry">
      <rect>
       <x>90</x>
       <y>40</y>
       <width>71</width>
       <height>23</height>
      </rect>
     </property>
     <property name="maximum">
      <number>1000</number>
     </property>
    </widget>
    <widget class="QCheckBox" name="randomJobs">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>0</y>
       <width>71</width>
       <height>16</height>
      </rect>
     </property>
     <property name="cursor">
      <cursorShape>PointingHandCursor</cursorShape>
     </property>
     <property name="text">
      <string>random</string>
     </property>
    </widget>
    <zorder>fromJobsLabel</zorder>
    <zorder>fromJobs</zorder>
    <zorder>toJobsLabel</zorder>
    <zorder>toJobs</zorder>
    <zorder>randomJobs</zorder>
    <zorder>constantJobs</zorder>
   </widget>
   <widget class="QGroupBox" name="researcherGroup">
    <property name="geometry">
     <rect>
      <x>210</x>
      <y>110</y>
      <width>171</width>
      <height>71</height>
     </rect>
    </property>
    <property name="title">
     <string>Researcher Budget</string>
    </property>
    <property name="checkable">
     <bool>false</bool>
    </property>
    <widget class="QDoubleSpinBox" name="constantResearcher">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>30</y>
       <width>71</width>
       <height>23</height>
      </rect>
     </property>
     <property name="showGroupSeparator" stdset="0">
      <bool>true</bool>
     </property>
     <property name="minimum">
      <double>1.000000000000000</double>
     </property>
     <property name="maximum">
      <double>1000.000000000000000</double>
     </property>
     <property name="singleStep">
      <double>0.010000000000000</double>
     </property>
    </widget>
    <widget class="QLabel" name="fromResearcherLabel">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>20</y>
       <width>28</width>
       <height>14</height>
      </rect>
     </property>
     <property name="text">
      <string>From</string>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="fromResearcher">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>40</y>
       <width>61</width>
       <height>23</height>
      </rect>
     </property>
     <property name="minimum">
      <double>1.000000000000000</double>
     </property>
     <property name="maximum">
      <double>1000.990000000000009</double>
     </property>
     <property name="singleStep">
      <double>0.010000000000000</double>
     </property>
    </widget>
    <widget class="QLabel" name="toResearcherLabel">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>20</y>
       <width>13</width>
       <height>14</height>
      </rect>
     </property>
     <property name="text">
      <string>To</string>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="toResearcher">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>40</y>
       <width>61</width>
       <height>23</height>
      </rect>
     </property>
     <property name="minimum">
      <double>1.000000000000000</double>
     </property>
     <property name="maximum">
      <double>1000.000000000000000</double>
     </property>
     <property name="singleStep">
      <double>0.010000000000000</double>
     </property>
    </widget>
    <widget class="QCheckBox" name="randomResearcher">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>0</y>
       <width>71</width>
       <height>16</height>
      </rect>
     </property>
     <property name="cursor">
      <cursorShape>PointingHandCursor</cursorShape>
     </property>
     <property name="text">
      <string>random</string>
     </property>
    </widget>
    <zorder>fromResearcherLabel</zorder>
    <zorder>fromResearcher</zorder>
    <zorder>toResearcherLabel</zorder>
    <zorder>toResearcher</zorder>
    <zorder>randomResearcher</zorder>
    <zorder>constantResearcher</zorder>
   </widget>
   <widget class="QGroupBox" name="studentGroup">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>110</y>
      <width>171</width>
      <height>71</height>
     </rect>
    </property>
    <property name="title">
     <string>Student Budget</string>
    </property>
    <property name="checkable">
     <bool>false</bool>
    </property>
    <widget class="QDoubleSpinBox" name="constantStudentBudget">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>30</y>
       <width>71</width>
       <height>23</height>
      </rect>
     </property>
     <property name="showGroupSeparator" stdset="0">
      <bool>true</bool>
     </property>
     <property name="minimum">
      <double>1.000000000000000</double>
     </property>
     <property name="maximum">
      <double>1000.000000000000000</double>
     </property>
     <property name="singleStep">
      <double>0.010000000000000</double>
     </property>
    </widget>
    <widget class="QLabel" name="fromStudentLabel">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>20</y>
       <width>28</width>
       <height>14</height>
      </rect>
     </property>
     <property name="text">
      <string>From</string>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="fromStudent">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>40</y>
       <width>61</width>
       <height>23</height>
      </rect>
     </property>
     <property name="minimum">
      <double>1.000000000000000</double>
     </property>
     <property name="maximum">
      <double>1000.000000000000000</double>
     </property>
     <property name="singleStep">
      <double>0.010000000000000</double>
     </property>
    </widget>
    <widget class="QLabel" name="toStudentLabel">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>20</y>
       <width>13</width>
       <height>14</height>
      </rect>
     </property>
     <property name="text">
      <string>To</string>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="toStudent">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>40</y>
       <width>61</width>
       <height>23</height>
      </rect>
     </property>
     <property name="minimum">
      <double>1.000000000000000</double>
     </property>
     <property name="maximum">
      <double>1000.000000000000000</double>
     </property>
     <property name="singleStep">
      <double>0.010000000000000</double>
     </property>
    </widget>
    <widget class="QCheckBox" name="randomStudent">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>0</y>
       <width>71</width>
       <height>16</height>
      </rect>
     </property>
     <property name="cursor">
      <cursorShape>PointingHandCursor</cursorShape>
     </property>
     <property name="text">
      <string>random</string>
     </property>
    </widget>
    <zorder>fromStudentLabel</zorder>
    <zorder>fromStudent</zorder>
    <zorder>toStudentLabel</zorder>
    <zorder>toStudent</zorder>
    <zorder>randomStudent</zorder>
    <zorder>constantStudentBudget</zorder>
   </widget>
   <widget class="QGroupBox" name="dateGroup">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>200</y>
      <width>171</width>
      <height>71</height>
     </rect>
    </property>
    <property name="title">
     <string>Starting Date</string>
    </property>
    <property name="checkable">
     <bool>false</bool>
    </property>
    <widget class="QCheckBox" name="nowDate">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>0</y>
       <width>61</width>
       <height>16</height>
      </rect>
     </property>
     <property name="cursor">
      <cursorShape>PointingHandCursor</cursorShape>
     </property>
     <property name="text">
      <string>now</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QDateTimeEdit" name="startDateTimeEdit">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>30</y>
       <width>141</width>
       <height>23</height>
      </rect>
     </property>
     <property name="displayFormat">
      <string>dd/MM/yyyy HH:mm:ss</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="requestsEndDate">
    <property name="geometry">
     <rect>
      <x>210</x>
      <y>200</y>
      <width>171</width>
      <height>71</height>
     </rect>
    </property>
    <property name="title">
     <string>Requests Span (seconds)</string>
    </property>
    <widget class="QSpinBox" name="requestsSpinBox">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>30</y>
       <width>81</width>
       <height>23</height>
      </rect>
     </property>
     <property name="minimum">
      <number>1</number>
     </property>
     <property name="maximum">
      <number>604800</number>
     </property>
     <property name="value">
      <number>10</number>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="nodesGroup">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>360</y>
      <width>111</width>
      <height>71</height>
     </rect>
    </property>
    <property name="title">
     <string>Number of nodes</string>
    </property>
    <widget class="QSpinBox" name="nodesSpinBox">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>30</y>
       <width>61</width>
       <height>23</height>
      </rect>
     </property>
     <property name="minimum">
      <number>128</number>
     </property>
     <property name="maximum">
      <number>512</number>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="coresGroup">
    <property name="geometry">
     <rect>
      <x>150</x>
      <y>360</y>
      <width>111</width>
      <height>71</height>
     </rect>
    </property>
    <property name="title">
     <string>Cores per node</string>
    </property>
    <widget class="QSpinBox" name="coresSpinBox">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>30</y>
       <width>61</width>
       <height>23</height>
      </rect>
     </property>
     <property name="minimum">
      <number>16</number>
     </property>
     <property name="maximum">
      <number>64</number>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="usagePriceGroup">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>280</y>
      <width>171</width>
      <height>71</height>
     </rect>
    </property>
    <property name="title">
     <string>Usage Price £ (per core)</string>
    </property>
    <widget class="QDoubleSpinBox" name="usagePriceSpinBox">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>30</y>
       <width>101</width>
       <height>23</height>
      </rect>
     </property>
     <property name="decimals">
      <number>6</number>
     </property>
     <property name="minimum">
      <double>0.000001000000000</double>
     </property>
     <property name="maximum">
      <double>5.000000000000000</double>
     </property>
     <property name="singleStep">
      <double>0.000001000000000</double>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="operationalCostGroup">
    <property name="geometry">
     <rect>
      <x>210</x>
      <y>280</y>
      <width>171</width>
      <height>71</height>
     </rect>
    </property>
    <property name="toolTip">
     <string/>
    </property>
    <property name="title">
     <string>Operational Cost £</string>
    </property>
    <widget class="QDoubleSpinBox" name="operationalCostSpinBox">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>30</y>
       <width>101</width>
       <height>23</height>
      </rect>
     </property>
     <property name="decimals">
      <number>6</number>
     </property>
     <property name="minimum">
      <double>0.000001000000000</double>
     </property>
     <property name="maximum">
      <double>5.000000000000000</double>
     </property>
     <property name="singleStep">
      <double>0.000001000000000</double>
     </property>
    </widget>
   </widget>
   <widget class="QTextBrowser" name="outputBox">
    <property name="geometry">
     <rect>
      <x>410</x>
      <y>40</y>
      <width>431</width>
      <height>381</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>600</x>
      <y>10</y>
      <width>41</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Output</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>863</width>
     <height>19</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuSupercomputer_Simulation">
    <property name="title">
     <string>Supercomputer Simulation</string>
    </property>
   </widget>
   <addaction name="menuSupercomputer_Simulation"/>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
